### ISSUE TITLE
unet keeps producing nan during training

### ISSUE THREAD

**Issue by EYcab** [kvablack/ddpo-pytorch#18]
![1705914514278](https://github.com/kvablack/ddpo-pytorch/assets/11568206/c728e1a6-9dd6-485b-9ec9-5ec52897f6b1)
Anyone knows why this unet process always produces nan results despite all the settings are done accordingly and all the other input variables are the same

**Comment by junyongyou** [comment]
Yes, I encountered the same here: loss becomes nan after some epochs. I tried different reward functions, and all the same.

**Comment by junyongyou** [comment]
I figured out the reason. You can change config.mixed_precision to "no" in base.py, such that full-precision can be enabled, and it should avoid that unet produces NaN.
